### PR TITLE
Don't create IR nodes for marker symbols

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - https://github.com/aspnet/Razor/issues/878
- Also setting the correct ContentLength from HtmlContentIRNode

@NTaylorMullen @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs [hunk 1]
```diff
@@ -181,11 +221,26 @@ public override void VisitStatementSpan(StatementChunkGenerator chunkGenerator,
 
             public override void VisitMarkupSpan(MarkupChunkGenerator chunkGenerator, Span span)
             {
+                if (span.Symbols.Count == 1 &&
+                    span.Symbols[0] is HtmlSymbol symbol &&
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs]: Bajillion pranav points. On another note though we should stay away from C# 7 until we move to it as a team 😄

File: src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs [hunk 2]
```diff
@@ -181,11 +221,26 @@ public override void VisitStatementSpan(StatementChunkGenerator chunkGenerator,
...
                 var currentChildren = Builder.Current.Children;
                 if (currentChildren.Count > 0 && currentChildren[currentChildren.Count - 1] is HtmlContentIRNode)
                 {
                     var existingHtmlContent = (HtmlContentIRNode)currentChildren[currentChildren.Count - 1];
                     existingHtmlContent.Content = string.Concat(existingHtmlContent.Content, span.Content);
+                    existingHtmlContent.SourceRange = new MappingLocation(
+                        existingHtmlContent.SourceRange.AbsoluteIndex,
+                        existingHtmlContent.SourceRange.LineIndex,
+                        existingHtmlContent.SourceRange.CharacterIndex,
+                        existingHtmlContent.SourceRange.ContentLength + span.Content.Length,
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs]: Amazing catch. 👍 👍 👍 👍

File: src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs [hunk 3]
```diff
@@ -126,7 +126,30 @@ public override void VisitStartTemplateBlock(TemplateBlockChunkGenerator chunkGe
 
             public override void VisitEndTemplateBlock(TemplateBlockChunkGenerator chunkGenerator, Block block)
             {
-                Builder.Pop();
+                var templateNode = Builder.Pop();
+                if (templateNode.Children.Count > 0)
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs]: Talked offline about this. We still want to ensure what the user wrote is flowed through the IR.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs]: Talked offline. Will remove node removal change but keeping the SourceRange setting.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Pull request merged**